src/en/credentials.md

```diff
@@ -13,57 +13,51 @@ We use the term *credentials* to describe the material necessary to do this
 
-Juju selects a credential according to how many credentials are defined. If you
-have only one credential, or if a credential is labelled 'default', then this
-is the credential that will be used by Juju. When multiple credentials are
-defined, with no default, a credential name must be specified at the model
-level.
-
+When credentials are added for a given cloud they become available to use on
+that cloud's controller and models. There are therefore two categories of
+credentials: those that are available (local) and those that have been sent to
+a controller (remote) due to requiring a credential for the first time.
+
+An active credential (one that has been used to bring about a change in Juju)
+is always associated with a cloud, a Juju user, and a model.
+ 
 ## Adding credentials
 
 Juju supports three methods for adding credentials:
 
- - Accepting credentials provided interactively by the user on the command line
- - Scanning for existing credentials via environment variables and/or "rc"
+ - accepting credentials provided interactively by the user on the command line
+ - scanning for existing credentials via environment variables and/or "rc"
    files (only supported by certain providers)
- - Reading a user-provided [YAML-formatted][yaml] file
+ - reading a user-provided [YAML-formatted][yaml] file
   
 !!! Note:
-    LXD deployments are a special case. Accessed from a Juju admin user, they
-    do not require credentials. Accessed from a non-admin user, a *certificate
-    credential* is needed. See
+    Local LXD deployments are a special case. Accessed from a Juju admin user,
+    they do not require credentials. Accessed from a non-admin user, a
+    *certificate credential* is needed. See
     [Additional LXD resources][clouds-lxd-resources-non-admin-creds] for
     details. 
 
 ### Adding credentials interactively
 
-You can add credentials interactively in this way:
+You can add credentials interactively. To do so with the AWS cloud:
 
 ```bash
-juju add-credential <cloud>
+juju add-credential aws
 ```
 
-You will be asked for credential information based on the chosen cloud. Here
-we're adding credentials for cloud 'aws':
+You will be asked for credential information based on the chosen cloud. For our
+AWS cloud the resulting interactive session would look like:
 
 ```no-highlight
 Enter credential name: carol
 Using auth-type "access-key".
 Enter access-key: *******
 Enter secret-key: *******
-Credentials added for cloud aws.
-```
-
-If you eventually set multiple credential names for the same cloud you will
-need to set one as the default:
-
-```bash
-juju set-default-credential <cloud> <credential-name>
 ```
 
-The default credential will be used when creating a controller with the
-`bootstrap` command. Otherwise, a credential can be specified with the
-`--credential` option with both the `bootstrap` and `add-model` commands.
+If you end up adding multiple credentials for the same cloud you will need to
```

Comment:
This is a bit trickier although very close to the truth...

The rule for selecting a credential is:
1. If there is only one, obviously use that one;
2. if there is more than one, select first one (I am *pretty* sure that this is not happening now but some older versions of Juju might still do that... so dunno if you want to test/mention it);
3. If there is more than one credential than a default one needs to be specified and juju will always use that default one unless the credential name is used explicitly, for example in 'add-model --credential'

Reply:
I only tested this with the beta.